platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,107 @@
+
+  @BeforeClass
+  public static void setupClass() throws Exception {
+
+    // read SPMetadata file into spMetadata
+    spMetadata = new ArrayList<>();
+    try (Stream<String> lines = Files.lines(Paths.get(FILE_PATH))) {
+      spMetadata.add(lines.collect(Collectors.joining(System.lineSeparator())));
+    }
+
+    // set up binding set
+    bindingSet = ImmutableSet.of(Binding.HTTP_POST, Binding.HTTP_REDIRECT, Binding.SOAP);
+  }
+
+  @Test
+  public void parseNullMetadata() throws Exception {
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(null, bindingSet);
+
+    assertThat("Returns an empty map", testMap, is(equalTo(EMPTY_MAP)));
+  }
+
+  @Test
+  public void parseCorrectMetadata() throws Exception {
+
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(spMetadata, bindingSet);
+
+    Map.Entry<String, EntityInformation> testEntry = testMap.entrySet().iterator().next();
+    String entityID = testEntry.getKey();
+    EntityInformation testInfo = testEntry.getValue();
+
+    assertThat("Entity ID is set", entityID, is(equalTo(ENTITY_ID)));
+    assertThat(
+        "Signing certificate is set", testInfo.getSigningCertificate(), is(equalTo(CERTIFICATE)));
+    assertThat(
+        "Encryption certificate is set",
+        testInfo.getEncryptionCertificate(),
+        is(equalTo(CERTIFICATE)));
+    assertThat(
+        "Redirect logout url is set",
+        testInfo.getLogoutService(Binding.HTTP_REDIRECT).getUrl(),
+        is(equalTo(LOGOUT_URL)));
+    assertThat(
+        "Post logout url is set",
+        testInfo.getLogoutService(Binding.HTTP_POST).getUrl(),
+        is(equalTo(LOGOUT_URL)));
+    assertThat(
+        "Redirect assertion consumer service url is set",
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_REDIRECT, 0).getUrl(),
+        is(equalTo(ACS_URL)));
+    assertThat(
+        "Post assertion consumer service url is set",
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_POST, 1).getUrl(),
+        is(equalTo(ACS_URL)));
+    assertThat(
+        "SOAP assertion consumer service url is set",
+        testInfo.getAssertionConsumerService(null, Binding.SOAP, 2).getUrl(),
```

Comment:
❓ This is just a question: In the SP's metadata we have an ACS for `PAOS` so would getting the one for `SOAP` return the one for `PAOS` or would it just not return the PAOS one cause it's not part of the supported binding? If it's the latter I suggest adding a check that the PAOS ACS is not returned.

❗ Also, it looks like if `getAssertionConsumerService` doesn't find the ACS for the binding you specified it just returns the default one so you should probably have a check for the binding that is returned not just the URL. I think you are getting the HTTP-Redirect url when you ask for SOAP.

Reply:
I made a change to use soap bindings and use unique urls to see which one is returned.